### PR TITLE
modsecurity_standalone: 2.9.3 -> 2.9.6

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Description of changes

Fixes CVE-2021-42717.

https://github.com/SpiderLabs/ModSecurity/releases/tag/v2.9.6
https://github.com/SpiderLabs/ModSecurity/releases/tag/v2.9.5
https://github.com/SpiderLabs/ModSecurity/releases/tag/v2.9.4

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>modsecurity_standalone</li>
  </ul>
</details>

A bit hard to test but the resulting content does not seem to be in worst shape than before the upgrade.

### CODE REVIEW

**Review by dasj19 — APPROVED** [review]
LGTM

File: pkgs/tools/security/modsecurity/default.nix [hunk 1]
```diff
@@ -37,7 +39,7 @@ stdenv.mkDerivation rec {
   outputs = ["out" "nginx"];
   # by default modsecurity's install script copies compiled output to httpd's modules folder
   # this patch removes those lines
-  patches = [ ./Makefile.in.patch ];
+  patches = [ ./Makefile.am.patch ];
```
Comment (risicle) [inline pkgs/tools/security/modsecurity/default.nix:43]:
```suggestion
  doCheck = true;
  checkInputs = [ perl ];
```

WFM linux and macos 10.15 and gives a little confidence in a package, let's be honest, neither of us are going to test fully.

Comment (LeSuisse) [reply pkgs/tools/security/modsecurity/default.nix:43]: Thanks for that!

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64-linux

**Pull request merged**